lib/puppet/type/cisco_portchannel_global.rb

```diff
@@ -0,0 +1,133 @@
+# Manages the Cisco PortChannel Global configuration resource.
+#
+# Dec 2015
+#
+# Copyright (c) 2013-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Puppet::Type.newtype(:cisco_portchannel_global) do
+  @doc = "
+    Manages the Cisco PortChannel Global configuration resource.
+    cisco_portchannel_global {'default':
+      ..attributes..
+    }
+    'default' is only acceptable name for this global config object.
+    Example:
+    cisco_portchannel_global { 'default':
```

Comment:
Sorry wrong place to put my comment. Please ignore my previous comment. 
These properties are special ones.
on n5 and n6k: only bundle_hash, bundle_select and hash_poly can be set and they all have to be set at the same time in one shot.
on n7k: only bundle_hash, bundle_select, rotate and asymmetric can be set and they all have to be set at the same time in one shot. load_defer and hash_distribution can be set separately and independently.
on3k and n9k: only bundle_hash, bundle_select, concatenation, rotate and symmetry can be set and they all have to be set at the same time in one shot. Resilient can be set separately and not dependent on others.